cmd/gtfTools/toBed.go

```diff
@@ -0,0 +1,101 @@
+package main
+
+import (
+	"flag"
+	"fmt"
+	"github.com/vertgenlab/gonomics/bed"
+	"github.com/vertgenlab/gonomics/chromInfo"
+	"github.com/vertgenlab/gonomics/exception"
+	"github.com/vertgenlab/gonomics/fileio"
+	"github.com/vertgenlab/gonomics/gtf"
+	"github.com/vertgenlab/gonomics/numbers/parse"
+	"log"
+	"os"
+	"strings"
+)
+
+// ToBedSettings defines the usage settings for the gtfTools toBed subcommand.
+type ToBedSettings struct {
+	InFile        string
+	OutFile       string
+	Tss           bool
+	ChromSizeFile string
+	Merge         bool
+}
+
+// toBedUsage defines the usage statement for the gtfTools toBed subcommand.
+func toBedUsage(toBedFlags *flag.FlagSet) {
+	fmt.Printf("gtfTools toBed - a tool to convert a gtf file into bed format.\n" +
+		"Usage:\n" +
+		"gtfTools toBed in.gtf out.bed\n" +
+		"options:\n")
+	toBedFlags.PrintDefaults()
+}
+
+// parseToBedArgs is the main function for the gtfTools toBed subcommand. It parses flags,
+// checks for errors, and executes the ToBed function.
+func parseToBedArgs() {
+	var expectedNumArgs int = 2
+	var err error
+	toBedFlags := flag.NewFlagSet("toBed", flag.ExitOnError)
+	var tss *bool = toBedFlags.Bool("tss", false, "Return a bed of tss positions annotated only with the geneName. Must provide chrom sizes file.")
+	var chromSizeFile *string = toBedFlags.String("chromSizeFile", "", "Specifies the name of a chrom.sizes file.")
+	var merge *bool = toBedFlags.Bool("merge", false, "Merge overlapping entries after converting all records to beds. Available with tss only.")
+	err = toBedFlags.Parse(os.Args[2:])
+	exception.PanicOnErr(err)
+	toBedFlags.Usage = func() { toBedUsage(toBedFlags) }
+	if len(toBedFlags.Args()) != expectedNumArgs {
+		toBedFlags.Usage()
+		log.Fatalf("Error: expected %d arguments, but got %d.\n", expectedNumArgs, len(toBedFlags.Args()))
+	}
+
+	inFile := toBedFlags.Arg(0)
+	outFile := toBedFlags.Arg(1)
+	s := ToBedSettings{
+		InFile:        inFile,
+		OutFile:       outFile,
+		Tss:           *tss,
+		ChromSizeFile: *chromSizeFile,
+		Merge:         *merge,
+	}
+
+	toBed(s)
+}
+
+// toBed parses an input gtf format file and converts each transcript into a bed entry.
+func toBed(s ToBedSettings) {
+	var line string
+	var doneReading bool
+	var nameString string
+	var currBed bed.Bed
+	var err error
+	if s.Tss && s.ChromSizeFile == "" {
+		log.Fatalf("Error: user must specify a chromSizes file to convert to a Tss bed.\n")
+	}
+
+	if s.Tss {
+		records := gtf.Read(s.InFile)
+		sizes := chromInfo.ReadToMap(s.ChromSizeFile)
+		beds := gtf.GenesToTssBed(records, sizes, s.Merge)
+		bed.Write(s.OutFile, beds)
+	} else {
+		file := fileio.EasyOpen(s.InFile)
+		out := fileio.EasyCreate(s.OutFile)
+		for line, doneReading = fileio.EasyNextRealLine(file); !doneReading; line, doneReading = fileio.EasyNextRealLine(file) {
+			words := strings.Split(line, "\t")
```

Comment:
When converting between file types, is it usually better to do strings.Split as opposed to first parsing into one struct (a gtf variable) and then indexing that struct to convert to another struct (a bed variable)? Or maybe it's better specifically in the case of gtfs?